crates/sp1/evm-exec/program/src/main.rs

```diff
@@ -87,8 +94,31 @@ pub fn main() {
             header.number
         ));
 
-        // TODO: Verify blob tx data equivalency against header.transactions_root
-        // https://github.com/celestiaorg/celestia-zkevm-hl-testnet/issues/68
+        let shares: Vec<Share> = blob_proof
+            .shares()
+            .iter()
+            .map(|b| Share::from_raw(b).expect("Failed to parse raw shares"))
+            .collect();
```

Comment:
I didn't realise that the blob_proof contained all the raw shares? Does that mean we are submitting the transactions twice: once in ClientExecutorInput and the other in each of the blob proofs? 

I thought what we would do is the other way around: Turn the transaction in the client executor input into blobs and assert that the blobs, given the blob proof, were part of the data root